sys/reaper/reaper_unix.go

```diff
@@ -38,17 +38,18 @@ const bufferSize = 2048
 func Reap() error {
 	now := time.Now()
 	exits, err := sys.Reap(false)
-	Default.Lock()
-	for c := range Default.subscribers {
-		for _, e := range exits {
-			c <- runc.Exit{
-				Timestamp: now,
-				Pid:       e.Pid,
-				Status:    e.Status,
-			}
+	for _, e := range exits {
+		done := Default.notify(runc.Exit{
+			Timestamp: now,
+			Pid:       e.Pid,
+			Status:    e.Status,
+		})
+
+		select {
+		case <-done:
+		case <-time.After(1 * time.Second):
```

Comment:
This seems to imply on a busy system that these notifications may get dropped.
Is there some way to record that this has happened (e.g. a log, a metric, etc)?

Reply:
These are coming from the shims so if you cannot read from a go channel in 1 sec then you have bigger issues than a dropped exit event

Reply:
If it is very unlikely to happen could it set err if `err == nil`? I noticed the error just gets logged anyway. I think this is not a good spot for a log or metric.